feature_engine/encoding/base_encoder.py

```diff
@@ -39,22 +39,32 @@ class BaseCategoricalTransformer(BaseEstimator, TransformerMixin):
     {variables}.
 
     {ignore_format}
+
+    allow_missing: bool
+        Whether to allow missing values in the columns handled by the transformer.
     """
 
     def __init__(
         self,
         variables: Union[None, int, str, List[Union[str, int]]] = None,
         ignore_format: bool = False,
+        allow_missing: bool = False
```

Comment:
If we add a parameter to this class, this parameter will be available to all encoders. Can we not add the param to the class that will use it only?

The least parameters in a class the better (more user friendly) so I would not want to add unneeded param to all transformers.

Reply:
Issue is, most of what the transformers do happens in the base class. If it were to be done outside of the base class, if would involve duplicating a lot of code.

Reply:
Can we not make this new class inherit the BaseCategorical and modify the functionality in the new class instead of in the base class? and if not, maybe make a child class of BaseCategorical and then modify that one instead?

BaseCategorical is used by all categorical transformers. So we can't really add a parameter that is only going to be used by one class. We need to find an alternative way.

Reply:
Issue is, since the part that differs is right in the middle of the base class' methods, adding a new subclass would involve duplicating lots of the code. Not sure if that's Ok for you.